pip-8210

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+if [[ ${-} != *i* ]]; then
+  source_once &> /dev/null && return 0
+fi
+
+#*# pip-8210
+
+#**
+# .. file:: pip-8210
+#
+# This script allows us to install a package outside of pip-tools. In order to accomplish correctly, we use the ``requirements.txt`` file as a constraint files,and this and handle corner cases like pip issue 8210.
+#
+# It add the -c constraints to ``pip install``, ``pip download``, and ``pip wheel``
+#
+# You should activate your virtualenv before calling this; it will use the ``pip`` on the path.
+#
+# :Arguments: * [--if-not-found <executable name>] - Only runs the ``pip`` command if a specified executable is not found.
```

Comment:
Should this say "runs the specified executable if the pip command is not found?

Reply:
No. It allows you to install `pip-tools` if `pip-sync` in not found. It's part of the chicken/egg resolution.

Reply:
Is this optional?

Reply:
Yes, there are `[]` for optional